### PR TITLE
fix species

### PR DESCRIPTION
**Pull request by Zebraside**

### CODE REVIEW

File: torchreid/data/datasets/image/dolphins.py [hunk 1]
```diff
@@ -67,30 +67,13 @@ def process_dir(self, dir_path, mode):
...
-                data.append((img_path, id, int(float(row['camera_id'])), row['species']))
+                data.append((img_path, int(row['id']), int(row['camera_id']), row['species_id']))
 
             if mode == 'train':
-                species_list = sorted(list(species_list))
-                species_map = {specie : i for i, specie in enumerate(species_list)}
-
                 for i, item in enumerate(data):
-                    data[i] = (*item[0:-1], 0, species_map[item[-1]]) # extra 0 is dataset id
+                    data[i] = (*item[0:-1], 0, item[-1]) # extra 0 is dataset id
```
Comment (sovrasov) [inline torchreid/data/datasets/image/dolphins.py]: We should do it by defaults for all the subsets

File: torchreid/data/datasets/image/dolphins.py [hunk 2]
```diff
@@ -67,30 +67,13 @@ def process_dir(self, dir_path, mode):
...
-                raw_id = int(float(row['id']))
-                if raw_id not in id_to_idx:
-                    id_to_idx[raw_id] = last_id
-                    last_id += 1
-                if mode == "train":
-                    id = id_to_idx[raw_id]
-                    species_list.add(row['species'])
-                else: id = raw_id
-                data.append((img_path, id, int(float(row['camera_id'])), row['species']))
+                data.append((img_path, int(row['id']), int(row['camera_id']), row['species_id']))
```
Comment (sovrasov) [inline torchreid/data/datasets/image/dolphins.py]: `int(row['species_id'])` just in case ...

Comment (Zebraside) [reply torchreid/data/datasets/image/dolphins.py]: fixed

File: torchreid/data/datasets/image/dolphins.py (hunk 1 above)
Comment (Zebraside) [reply torchreid/data/datasets/image/dolphins.py]: condition removed

**Review by sovrasov — APPROVED** [review]

**Pull request merged**